### DISCUSSION TITLE
Home Assistant Discovery funktioniert bei mir nicht

### DISCUSSION THREAD

**Discussion by Willi24c** [softwarecrash/EPEver2MQTT#53] [General]
Ich verwende HA 2024.9.2. Der MQTT Explorer in HA zeigt mir die Daten / Informationen von "epever EP1" meines XTRA4210N an also gehe ich davon aus, dass die Konfiguration des EPEver2MQTT in Ordnung ist. In der HA Devices List wird das EPEver2MQTT nicht angezeigt. Ich habe nun versucht, in der configuration.yaml einen sensor anzulegen:

![Screenshot 2024-09-22 212749](https://github.com/user-attachments/assets/2b4ccadd-867d-43ef-a4ba-a4b5e3cd3869)

Diesen Sensor kann ich bei den Entitäten finden, aber da gibt es keinen Wert (Unbekannt)

![Screenshot 2024-09-22 213006](https://github.com/user-attachments/assets/3435f1c8-8448-40e0-a8cc-9b3dfd891292)

Diese manuelle Konfiguration sollte ja eigentlich nicht notwendig sein, hab ich in meiner Not aber mal so probiert.

Es ist zu erwähnen, das ich die Software "Solar Assistant" verwende. Für dieses Gerät finde ich bei Integrationen ein MQTT Device und die Verbindung funktioniert einwandfrei. Bei Solar Assistant.io gibt es eine Anleitung für MQTT, das hat so auf Anhieb funktioniert. Ein Eintrag in der configuration.yaml war nicht notwendig.

![Screenshot 2024-09-22 213153](https://github.com/user-attachments/assets/b8997819-7535-41e5-bcf9-319ea679203b)

Wenn noch weitere Informationen zu meinem System oder den Einstellungen erforderlich sind, so lass es mich wissen.

Vielen Dank im Voraus für eine Hilfestellung oder Idee, wo ich noch nachsehen soll.

Mit freundlichen Grüßen aus Hessen
Harald

**Comment by all-solutions** [comment]
Du hast ein "Doppel-Topic" (epever/EP_1/) angegeben. Daher funktioniert es nicht.

Probier mal z.B. nur Epever_1 als Topic in Epever2MQTT, dann solltest Du alles in HA sehen.

**Reply by Willi24c** [reply]
Mir ist nicht ganz klar wo ich das ändern soll.
![Screenshot 2024-09-22 233044](https://github.com/user-attachments/assets/c593eeba-637a-47ed-82b8-b60fc46d9d8b)
Ich hatte die Eingaben so wie im Beispiel gemacht (siehe Bild). 
Soweit ich das verstanden habe setzt sich das (epever/EP_1) aus der Device Quantity und dem MQTT Topic zusammen.
Ist Epever_1  bei MQTT Topic dann die Lösung?

MfG
Harald

**Reply by all-solutions** [reply]
Ne, warte. Da war ich etwas auf dem falschen Weg. Sorry, das war mein Fehler.

Ich schau mir das Morgen genauer an. 
Hast Du den MQTT-Explorer auf dem Rechner? Wenn ja, schau mal bitte, ob auf dem Broker im Topic homeassistant etwas über den EP_1 zu sehen ist.

**Comment by Willi24c** [comment]
Ja, siehe Bild
![Screenshot 2024-09-22 234714](https://github.com/user-attachments/assets/b69ac0ee-d10f-44d2-9074-0b4d3a68e03f)

Das HA Topic für die configuration.yaml hatte ich da raus kopiert.

![Screenshot 2024-09-22 235032](https://github.com/user-attachments/assets/af82e0ab-ade7-4200-953e-9508da26b428)


MfG
Harald

**Reply by softwarecrash** [reply]
eigentlich brauchst du garkeine configuration mehr, das macht der esp i.d.r. selbst

**Reply by Willi24c** [reply]
Habe ich auch gedacht, funktioniert bei mir aber nicht. Deshalb habe ich wie oben geschrieben, versucht über die configuration.yaml weiterzukommen. Wie erwähnt, hat das Auto-Discovery bei SolarAssistant einwandfrei funktioniert.

Wie auch immer, ich bin für jeden Hinweis dankbar.
Harald

**Reply by softwarecrash** [reply]
unter homeassistant/sensor, nimm dir mal einen beliebigen wert, da drinne steht ein mqtt pfad zum eigentlichen datenpunkt, stimmt der mit deinem baum überein?

**Reply by Willi24c** [reply]
Hallo,
ich versuche das mal für mich zu übersetzten:
1.) MQTT Explorer sensor:

![Screenshot 2024-09-23 175422](https://github.com/user-attachments/assets/10aa7251-e773-4473-9a37-0e57e0257cf4)

2.) MQTT Explorer sensor Details:

![Screenshot 2024-09-23 175525](https://github.com/user-attachments/assets/f91cd570-6bfc-492b-b027-faf01f04afa9)

3.) Mein Eintrag in configuration.yaml (den ich ja eigentlich nicht brauche:
![Screenshot 2024-09-23 175714](https://github.com/user-attachments/assets/729348e6-694f-4a52-bdc4-04ecabb232ab)

Ist es das was du sehen möchtest?

MfG
Harald

**Reply by softwarecrash** [reply]
ja, und das sieht eigentlich auch so aus wie es soll.
evt spinnt HA rum, wäre nicht das erste mal

**Reply by Willi24c** [reply]
Ich habe ja schon erwähnt, dass ich eine MQTT Verbindung zu SolarAssistant hergestellt hatte. Da gab es auch keinen Eintrag in der cofiguration.yaml. Dafür wurde aber ein Gerät unter Integrationen angelegt. Da finde ich alle Werte von meinem Voltronic Invertern und den Pylontech Batterien. 
Kann es sein, das es da einen Zusammenhang gibt (die vorhandene MQTT Integration irgendwie stört)?

Ich bin kein Software Spezialist, daher bin ich hier auf eure Hilfe angewiesen.

Danke im Voraus für eure Hilfestellung.
Harald

**Reply by softwarecrash** [reply]
vielleicht kann @jarnsen da noch weiterhelfen

**Comment by Jarnsen** [comment]
Schmeiß den Sensor aus der yaml raus, wir müssen ja rausfinden warum im Explorer angezeigt wird aber nicht unter mqtt. 

Außer dem ist der Sensor unvollständig und hat den falschen Pfad. Ich habe da so ne Idee, wir prüfen die gerade intern. 

Lass mich raten, Load_V wird auch nicht angezeigt

**Reply by Willi24c** [reply]
Hi Jarnsen,
ist auskommentiert und HA Neustart. 
Den letzten Satz habe ich wohl nicht richtig verstanden, wie auch immer, im MQTT Explorer wird Load_V mit 26.26V angezeigt (ist eine 24V LiFePo Batterie). 

MfG
Harald

**Reply by Jarnsen** [reply]
ok, was mir spontan aufgefallen ist, ist das beim Ursprungspfad 
![image](https://github.com/user-attachments/assets/ba2f45da-3da4-4a28-bfa7-834c82d1cc85)
oft ein Unterstrich fehlt, z.B. BATT_V, bei dir steht aber BATT V, kann sein das es das schon ausmacht. Ebenso würde ich dem MQTTSERVER immer mit der IP arbeiten und nicht mit HASS.IO.

**Wenn du möchtest kannst du mir gerne privat über Discord schreiben, dann kann ich mir das mit TEAMVIEWER mal anschauen, wenn du das willst.**

PS: gerade habe ich nochmal alles durchgespielt, alles unter homeassistant/sensor mit EPEVER gelöscht und wieder durch nen reboot des ESP erstellt, läuift ohne Probleme.

**Reply by Willi24c** [reply]
Hallo Jarnsen,
vielen Dank für deine Nachricht und Vorschläge. Ich war zwei Tage in einer Klinik für einen kleinen Eingriff und konnte daher nicht schneller antworten.
Das mit dem fehlenden Unterstrich habe ich nicht ganz verstanden. Bei mir stand bei Batt_V immer ein Unterstrich, das ist wohl im Bild nicht richtig erkennbar. Ich habe schnell mal bei dem MQTT Explorer auf IP umgestellt - keine Änderung. Oder meinst du den Mosquitto Broker, da weiß ich nicht ob oder wie man den umstellen kann.

Wo hast du alles "unter homeassistant/sensor" gelöscht? Im MQTT Explorer? Hab ich gerade mal gemacht, alle Details im Explorer sind nach dem ESP Restart wieder aufgetaucht, Aber bei Home Assistant hat das nicht geholfen. 

Teamviewer hab ich nicht, aber bei Anydesk gibt es eine Free Version, die könnte ich installieren.

Gruß
Harald

**Reply by Willi24c** [reply]
Hallo Jarnsen,
ich habe die kostenlose TeamViewer Version gefunden und auf einem MAC installiert.
Bitte gib mir Bescheid ob und wann wir eine Session starten.
Gruß
Harald

**Reply by Jarnsen** [reply]
lass uns den Rest über Discord per PN machen

**Reply by Willi24c** [reply]
Hallo Jarnsen,
ich habe Discord zwar installiert aber das ist total unübersichtlich (habe ich bisher noch nie benutzt) und daher bin ich von der für mich  mühsamen MQTT Version auf eine ESPHome WIFI Variante (aktuell von Chickey) umgestiegen. Das funktioniert prima.
Danke noch für (eure) deine Mühe.
